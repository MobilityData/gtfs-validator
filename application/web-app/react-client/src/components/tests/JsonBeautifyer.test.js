/*
 *  Copyright (c) 2020. MobilityData IO.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import React from "react";
import JsonBeautyfier from "../JsonBeautyfier";
import renderer from "react-test-renderer";

describe("JsonBeautyfier", () => {
    it("should be defined", () => {
        expect(JsonBeautyfier).toBeDefined();
    });

    it("should render correctly without data", () => {
        const tree = renderer.create(
            <JsonBeautyfier />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("should render correctly with data", () => {
        let mockData = {
            "key1": "value1",
            "key2": "value2"
        }
        const tree = renderer.create(
            <JsonBeautyfier data={mockData}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
