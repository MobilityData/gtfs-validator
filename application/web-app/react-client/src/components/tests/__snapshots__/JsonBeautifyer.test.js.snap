// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonBeautyfier should render correctly with data 1`] = `
<pre
  style={
    Object {
      "backgroundColor": "rgb(39, 40, 34)",
      "overflow": "auto",
    }
  }
>
  <span
    style={
      Object {
        "color": "rgb(129, 211, 204)",
      }
    }
  >
    {
  </span>
  <div>
    <div>
      <span>
          
      </span>
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        "
        <span
          style={
            Object {
              "color": "rgb(201, 70, 56)",
            }
          }
        >
          key1
        </span>
        "
        <span
          style={
            Object {
              "color": "rgb(129, 211, 204)",
            }
          }
        >
          :
        </span>
         
      </span>
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        "
        <span
          style={
            Object {
              "color": "rgb(242, 152, 59)",
            }
          }
        >
          value1
        </span>
        "
      </span>
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        ,
      </span>
    </div>
    <div>
      <span>
          
      </span>
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        "
        <span
          style={
            Object {
              "color": "rgb(201, 70, 56)",
            }
          }
        >
          key2
        </span>
        "
        <span
          style={
            Object {
              "color": "rgb(129, 211, 204)",
            }
          }
        >
          :
        </span>
         
      </span>
      <span
        style={
          Object {
            "color": "rgb(129, 211, 204)",
          }
        }
      >
        "
        <span
          style={
            Object {
              "color": "rgb(242, 152, 59)",
            }
          }
        >
          value2
        </span>
        "
      </span>
      
    </div>
  </div>
  <span
    style={
      Object {
        "color": "rgb(129, 211, 204)",
      }
    }
  >
    
    }
  </span>
</pre>
`;

exports[`JsonBeautyfier should render correctly without data 1`] = `
<pre
  style={
    Object {
      "backgroundColor": "rgb(39, 40, 34)",
      "overflow": "auto",
    }
  }
>
  <span
    style={
      Object {
        "color": "rgb(201, 70, 56)",
      }
    }
  >
    null
  </span>
</pre>
`;
